### PR TITLE
Enable analysis on non-Windows OSes

### PR DESCRIPTION
**Pull request by michalb-sonar**
Fix #944

### CODE REVIEW

File: sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java [hunk 1]
```diff
@@ -74,11 +74,6 @@ private boolean shouldExecuteOnProject(FileSystem fs) {
       return false;
     }
 
-    if (!system.isOsWindows()) {
-      throw MessageException.of("C# analysis is not supported on " + System.getProperty("os.name") +
-        ". Please, refer to the SonarC# documentation page for more information.");
-    }
-
     return true;
```
Comment (SonarTech) [inline sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java]: MAJOR Make "shouldExecuteOnProject" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CSharpSensor.java#L55](https://github.com/SonarSource/sonar-csharp/blob/7c2d627c4f4ae644ace99900335f7de018cdc0f6/sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java#L55): Remove this unused "system" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. ![MINOR][MINOR] [CSharpSensor.java#L27](https://github.com/SonarSource/sonar-csharp/blob/7c2d627c4f4ae644ace99900335f7de018cdc0f6/sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java#L27): Remove this unused import 'org.apache.commons.lang.SystemUtils'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [CSharpSensor.java#L38](https://github.com/SonarSource/sonar-csharp/blob/7c2d627c4f4ae644ace99900335f7de018cdc0f6/sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java#L38): Remove this unused import 'org.sonar.api.utils.MessageException'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSensorTest.java [hunk 2]
```diff
@@ -125,20 +125,6 @@ public void noAnalysisIfNoFilesDetected() throws Exception {
...
-
-    // Assert exception
-    thrown.expect(MessageException.class);
-    thrown.expectMessage("C# analysis is not supported");
-
-    // Act
-    sensor.execute(tester);
-  }
-
-  @Test
```
Comment (Evangelink) [inline sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSensorTest.java]: Something's weird here. Either you have removed an attribute you shouldn't have or you forgot to remove a method.

File: sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java [hunk 3]
```diff
@@ -74,11 +74,6 @@ private boolean shouldExecuteOnProject(FileSystem fs) {
       return false;
     }
 
-    if (!system.isOsWindows()) {
```
Comment (valhristov) [inline sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java]: I remember adding this check recently ;)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CSharpSensor.java#L48](https://github.com/SonarSource/sonar-csharp/blob/7c2d627c4f4ae644ace99900335f7de018cdc0f6/sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java#L48): Remove this unused "system" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [CSharpSensor.java#L71](https://github.com/SonarSource/sonar-csharp/blob/7c2d627c4f4ae644ace99900335f7de018cdc0f6/sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java#L71): Make "shouldExecuteOnProject" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MINOR][MINOR] [CSharpSensor.java#L34](https://github.com/SonarSource/sonar-csharp/blob/7c2d627c4f4ae644ace99900335f7de018cdc0f6/sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java#L34): Remove this unused import 'org.sonar.api.utils.MessageException'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSensorTest.java (hunk 2 above)
Comment (Evangelink) [reply sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSensorTest.java]: Forget about it....I haven't seen top `@Test` wasn't removed.

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CSharpSensor.java#L55](https://github.com/SonarSource/sonar-csharp/blob/1534aa27c1e953e0dedf51f52f87f26cb309d22e/sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java#L55): Remove this unused "system" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [CSharpSensor.java#L77](https://github.com/SonarSource/sonar-csharp/blob/1534aa27c1e953e0dedf51f52f87f26cb309d22e/sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java#L77): Make "shouldExecuteOnProject" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MINOR][MINOR] [CSharpSensor.java#L27](https://github.com/SonarSource/sonar-csharp/blob/1534aa27c1e953e0dedf51f52f87f26cb309d22e/sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java#L27): Remove this unused import 'org.apache.commons.lang.SystemUtils'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [CSharpSensor.java#L38](https://github.com/SonarSource/sonar-csharp/blob/1534aa27c1e953e0dedf51f52f87f26cb309d22e/sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java#L38): Remove this unused import 'org.sonar.api.utils.MessageException'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java [hunk 4]
```diff
@@ -24,7 +24,7 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import org.apache.commons.lang.SystemUtils;
+import java.util.Optional;
```
Comment (SonarTech) [inline sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java]: MINOR Remove this unused import 'java.util.Optional'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**